app/domain/repository/persist_repository_from_upstream.rb

```diff
@@ -111,13 +111,15 @@ def self.correct_status_from_upstream(repository, archived_upstream:)
                             })
       # set to unmaintained if we do not have another status already assigned
       "Unmaintained"
-    elsif !archived_upstream && repository.status == "Unmaintained"
-      StructuredLog.capture("REPOSITORY_REMOVE_UNMAINTAINED_STATUS",
+    elsif !archived_upstream && %w[Unmaintained Removed].include?(repository.status)
+      StructuredLog.capture("REPOSITORY_UNSETTING_REMOVED_STATUS",
```

Comment:
```suggestion
      StructuredLog.capture("REPOSITORY_UNSETTING_STATUS",
```
I know the previous comment suggested this but maybe remove the `REMOVED_STATUS` naming and have `REPOSITORY_UNSETTING_STATUS` because it could be either Removed or Unmaintained being unset here. Minor detail so feel free to ignore since we will likely be searching by whatever is string is set here.